### ISSUE TITLE
[BUG] [T6] A problem with setting struct elements

### ISSUE THREAD

**Issue by Zi0MIX** [xensik/gsc-tool#143]
I've encountered a problem with recent releases related to structs. Directly updating struct elements seems to no longer be possible, unless it's done via `self.` (just the very presence of self. in front of the line helps, as you can see with the examples below). Problem is present on versions 1.1.0 and 1.0.688

I will mention that examples of the code that no longer work, do work just fine with version 1.0.540 as well as old compiler. Examples are difectly from my code base. I tried to provide just the relevant parts to keep it compact, but if more details are needed i can also do that

Example 1:
``` cpp
    watermark = createserverfontstring("hudsmall" , 1.2);
    watermark setPoint("CENTER", "TOP", x_pos, -5);
    watermark.color = color;
    watermark setText(text);
    watermark.alpha = alpha_override;
    watermark.hidewheninmenu = 0;
```
does not update eiter color or alpha, but as soon as the update is abstracted to a separate function, it works just fine
``` cpp
    watermark = createserverfontstring("hudsmall" , 1.2);
    watermark setPoint("CENTER", "TOP", x_pos, -5);
    watermark set_hud_color(color);
    // watermark.color = color;
    watermark setText(text);
    watermark set_hud_alpha(alpha_override);
    // watermark.alpha = alpha_override;
    watermark.hidewheninmenu = 0;

set_hud_color(new_color)
{
    self.color = new_color;
}

set_hud_alpha(new_alpha)
{
    self.alpha = new_alpha;
}
```

Example 2:
``` cpp
        self.hud_velocity = createfontstring("default" , 1.1);
	self.hud_velocity set_hud_properties("hud_velocity", "CENTER", "CENTER", "CENTER", 200);
	self.hud_velocity.alpha = 0.75;
	self.hud_velocity.hidewheninmenu = 1;

    while (true)
    {
        self velocity_visible(self.hud_velocity);

	velocity = int(length(self getvelocity() * (1, 1, 0)));
	self.hud_velocity velocity_meter_scale(velocity);
        self.hud_velocity setValue(velocity);

        wait 0.05;
    }

velocity_visible(hud)
{
    if (is_true(self.afterlife) || getDvar("velocity") == "0")
        hud.alpha = 0;
    else
        hud.alpha = 1;
}
```
Leaves alpha element permanently at `0.75`, but changing it into something like that:
``` cpp
        self.hud_velocity = createfontstring("default" , 1.1);
	self.hud_velocity set_hud_properties("hud_velocity", "CENTER", "CENTER", "CENTER", 200);
	self.hud_velocity.alpha = 0.75;
	self.hud_velocity.hidewheninmenu = 1;

    while (true)
    {
        self.hud_velocity.alpha = self velocity_visible();

	velocity = int(length(self getvelocity() * (1, 1, 0)));
	self.hud_velocity velocity_meter_scale(velocity);
        self.hud_velocity setValue(velocity);

        wait 0.05;
    }
}

velocity_visible(default_alpha)
{
    if (!isDefined(default_alpha))
        default_alpha = 0.75;

    if (is_true(self.afterlife) || getDvar("velocity") == "0")
        return 0;
    return default_alpha;
}
```
Performs the update just fine. The 2nd example does it completely the opposite way of the first one, but still the only way it works is to update the element via `self.`

**Comment by diamante0018** [comment]
It is possible this was a regression.
You pointed out this used to work, I will try to look a this myself since Xensik is mostly unavailable at this time but he is able to merge fix pull requests.

**Comment by diamante0018** [comment]
I could not find any difference between 1.0.540 and 'dev'.
**There is no difference in the assembly output.**
This is most likely a user error, subtle differences between using self and not using it can break the code.
After all, if there was a compiler issue, your described workaround should prove to still be not working. It points to some other issue.
This does not appear to be a compiler issue

**Comment by Zi0MIX** [comment]
Well i tested the exact same code across all 5 releases available on GitHub, and 1.0.540 was the only one that worked. A friend of mine compiled your current dev branch for me, and it works again on that. I ran disassembly mode on scripts produced by all 3 (1.0.540, 1.1.0 and dev) and 1.1.0 is missing `OP_CastFieldObject()` call around the code that's misbehaving. So yeah i'm pretty certain it is a compiler issue, however a fix seems to be already implemented so the only thing that's needed to solve this issue is a new release. I'll leave it open for now, in case somebody finds themselves in similar situation it's easier to find open, unless you think otherwise, then feel free to close it already. I'll otherwise close the issue after a new release is out and working.

**Comment by xensik** [comment]
#141
